app/models/ta_membership.rb

```diff
@@ -30,22 +31,24 @@ def self.from_csv(assignment, csv_data, remove_existing)
 
       row.drop(1).each do |grader_name|
         unless graders[grader_name].nil?
-          new_ta_memberships << TaMembership.new(
-            grouping_id: groupings[row[0]],
-            user_id: graders[grader_name]
-          )
+          new_ta_memberships_hash.push({user_id: graders[grader_name], grouping_id: groupings[row[0]],
+                                        type: "TaMembership"})
         end
       end
     end
+
     Repository.get_class.update_permissions_after do
-      TaMembership.import new_ta_memberships, validate: false, on_duplicate_key_ignore: true
+      unless new_ta_memberships_hash.empty?
+        TaMembership.insert_all(new_ta_memberships_hash)
+      end
     end
 
+
     # Recompute criteria associations
     if assignment.assign_graders_to_criteria
       Grouping.update_criteria_coverage_counts(
         assignment,
-        new_ta_memberships.map { |x| x[:grouping_id] }
+        new_ta_memberships_hash.map{|row| row[:grouping_id]}
```

Comment:
In case you aren't sure about what this means, you should include blanks spaces inside `{ ... }` (this is a different style than what we used in Python).